### PR TITLE
[WIP] Support HTTP proxy settings in windows

### PR DESCRIPTION
**Pull request by msabansal**
**- What I did**
This PR adds support of applying http proxy settings for windows containers. This feature was added for RS4 and supports setting proxies as documented at https://docs.docker.com/network/proxy/

Also revendor hcsshim v0.6.13 required for this change.
**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: libcontainerd/client_local_windows.go [hunk 1]
```diff
@@ -299,6 +300,78 @@ func (c *client) createWindows(id string, spec *specs.Spec, runtimeOptions inter
 		}
 	}
 	configuration.MappedDirectories = mds
+
+	// Proxy Support for RS4
+	if system.GetOSVersion().Build > 17120 {
```
Comment (jhowardmsft) [inline libcontainerd/client_local_windows.go]: >=17134?

File: libcontainerd/client_local_windows.go [hunk 2]
```diff
@@ -299,6 +300,78 @@ func (c *client) createWindows(id string, spec *specs.Spec, runtimeOptions inter
...
+				strings.HasPrefix(envVar, "FTP_PROXY") {
+				proxy = strings.Split(envVar, "=")[1]
+			} else if strings.HasPrefix(envVar, "NO_PROXY") {
+				exceptions = strings.Replace(strings.Split(envVar, "=")[1], ",", ";", -1)
+			}
+		}
+
+		if proxy != "" {
+			defaultConnectionString := []byte{
+				0x46, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
```
Comment (jhowardmsft) [inline libcontainerd/client_local_windows.go]: What are these? Is there an MSDN link or something? (More for interest, not questioning correctness). If there's documentation, might be worth adding a link to it.

File: libcontainerd/client_local_windows.go (hunk 1 above)
Comment (msabansal) [reply libcontainerd/client_local_windows.go]: replaced the check.

File: libcontainerd/client_local_windows.go (hunk 2 above)
Comment (msabansal) [reply libcontainerd/client_local_windows.go]: At the time I added these I remember having used MSDN for figuring this out. I could find these documents https://support.microsoft.com/en-us/help/900935/how-the-windows-update-client-determines-which-proxy-server-to-use-to and https://securelink.be/blog/windows-proxy-settings-explained/ but didn't find any documents for the exact registry settings documented

**Comment by daschott** [comment]
@thaJeztah can you please take a look at this? The PR contains an RS4+ (Windows 1803 and above) feature. How are all the tests going to pass if the CI uses RS1?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Left some comments inline.

Even though we don't have RS4 nodes running in CI currently (I think works was in progress for this); it may be worth preparing a test for this (but skipped on older versions)

**Comment by fntlnz** [comment]
@msabansal are you still moving this forward?

**Comment by daschott** [comment]
@fntlnz Sandeep switched teams recently, but we someone else will take a look at addressing the comments soon.

**Comment by olljanat** [comment]
@msabansal this one needs to be rebased

**Comment by olljanat** [comment]
Derek add label: status/failing-ci

**Comment by thaJeztah** [comment]
ping @msabansal @jhowardmsft what's the status on this one?

**Comment by ddebroy** [comment]
@ameyag

**Comment by ameyag** [comment]
PR corresponding to the vendored change has been closed and not merged. Microsoft/hcsshim#163

**Comment by thaJeztah** [comment]
@ameyag looks like was carried in https://github.com/microsoft/hcsshim/pull/259 for the v0.6.12 branch, which also got closed, but ultimately merged as https://github.com/microsoft/hcsshim/pull/272

**Comment by thaJeztah** [comment]
That said, I think those magic values should ultimately be somewhere in hcsshim (as constants, a utility function or whatever makes most sense)

**Comment by thaJeztah** [comment]
@ameyag I'll close this one

**Pull request closed without merging**